### PR TITLE
states for various gestures added

### PR DESCRIPTION
**Pull request by enchev**
Implemented #1048

### CODE REVIEW

**Comment by enchev** [comment]
run ci

File: ui/gestures/gestures.d.ts [hunk 1]
```diff
@@ -40,6 +40,28 @@ declare module "ui/gestures" {
...
+     */
+    export enum GestureStateTypes {
+        /**
+         * Gesture cancelled.
+         */
+        cancelled,
+        /**
+         * Gesture began.
+         */
+        began,
```
Comment (hshristov) [inline ui/gestures/gestures.d.ts]: Why not using - start, end, change, cancel?

**Comment by hshristov** [comment]
:+1:

**Comment by rclai** [comment]
What is the API for tap down and up? Also, how does this look on XML?

**Comment by enchev** [comment]
I've just updated our docs with examples how to use gestures with state (tap, double tap and long press gestures are discreet). Here is an example with pan gesture:
https://github.com/NativeScript/docs/blob/production/gestures.md#pan

**Pull request merged**